src/script/deploy.js

```diff
@@ -213,16 +277,27 @@ async function deployLineaWorldID(config) {
   const spinner = ora("Deploying LineaID on Linea...").start();
 
   try {
-    const data = execSync(
-      `forge script src/script/DeployLineaWorldID.s.sol:DeployOpWorldID --fork-url ${config.lineaRpcUrl} \
-      --etherscan-api-key ${config.lineaEtherscanApiKey} --broadcast --verify -vvvv`,
-    );
-    console.log(data.toString());
+    let command = `forge script src/script/DeployLineaWorldID.s.sol:DeployLineaWorldID --fork-url ${config.lineaRpcUrl} --broadcast --json`;
+    if (config.lineaEtherscanApiKey) {
+      command += ` --etherscan-api-key ${config.lineaEtherscanApiKey} --verify`;
+    }
+    const output = execSync(command);
+    const data = output.toString();
+    const jsonData = parseJson(data);
+    if (jsonData.success) {
+      for (const log of jsonData.logs) {
+        if (!log.includes("LineaWorldID")) continue;
+        const match = data.match(addressRegex);
+        if (!match) continue;
+        const contractAddress = match[0];
+        config.lineaWorldIDAddress = contractAddress;
+      }
```

Comment:
That still looks like parsing lots of raw logs, json didn't help us to simplify the data extraction 🤔.
I was looking for a better way of "returning" data from Forge script originally. Do you think that log parsing is still the best one?

Reply:
![reply](https://github.com/user-attachments/assets/4fef3c7d-3d8f-4c06-98ec-1bae24a13da3)

The --json flag outputs the transaction logs in JSON format to the std out of the terminal.

The Issue stated the question on how this logs "output provided by the --json flag" could be used to update the config.json provided the Forge version is pinned and the output from the --json flag remains consistent.

So from parsing this logs we would then be able to update and store the deployed contract addresses.